e2e-tests/gef/cypress/e2e/clone/facility-end-date-enabled/clone-v0-to-v1.spec.js

```diff
@@ -0,0 +1,109 @@
+
+      cy.visit(relative(`/gef/application-details/${version0DealId}`));
+
+      cloneGEFDeal.cloneGefDealLink().click();
+      mandatoryCriteria.trueRadio().click();
+
+      cy.clickContinueButton();
+
+      cy.keyboardInput(nameApplication.internalRef(), 'Cloned of v0 deal');
+      cy.clickContinueButton();
+
+      cy.get('table.govuk-table tr').eq(1).find('td').eq(1).find('.govuk-link').click();
+
+      cy.getDealIdFromUrl().then((id) => {
+        clonedDealId = id;
+      });
+    });
+
+    beforeEach(() => {
+      cy.login(BANK1_MAKER1);
+
+      cy.visit(relative(`/gef/application-details/${clonedDealId}`));
+    });
+
+    it('shows required for has bank provided facilityEndDate', () => {
+      applicationDetails.facilitySummaryListTable(0).isUsingFacilityEndDateValue().should('contain', 'Required');
+      submitButton().should('not.exist');
+    });
+
+    it('isUsingFacilityEndDate is unset on about this facility page', () => {
+      applicationDetails.facilitySummaryListTable(0).isUsingFacilityEndDateAction().click();
+
+      aboutFacilityUnissued.isUsingFacilityEndDateYes().should('not.be.checked');
+      aboutFacilityUnissued.isUsingFacilityEndDateNo().should('not.be.checked');
+    });
+
+    it('facility table renders "required" for facilityEndDate after isUsingFacilityEndDate is set to true', () => {
+      applicationDetails.facilitySummaryListTable(0).isUsingFacilityEndDateAction().click();
+      aboutFacilityUnissued.isUsingFacilityEndDateYes().click();
+      cy.clickSaveAndReturnButton();
+
+      applicationDetails.facilitySummaryListTable(0).facilityEndDateValue().should('contain', 'Required');
+      submitButton().should('not.exist');
+    });
+
+    it('facility end date page does not show pre-populated values', () => {
+      applicationDetails.facilitySummaryListTable(0).facilityEndDateAction().click();
+
+      facilityEndDate.facilityEndDateDay().should('have.value', '');
+      facilityEndDate.facilityEndDateMonth().should('have.value', '');
+      facilityEndDate.facilityEndDateYear().should('have.value', '');
+    });
+
+    it('facility table renders "required" for bankReviewDate after isUsingFacilityEndDate is set to false', () => {
+      applicationDetails.facilitySummaryListTable(0).isUsingFacilityEndDateAction().click();
+      aboutFacilityUnissued.isUsingFacilityEndDateNo().click();
+      cy.clickSaveAndReturnButton();
+
+      applicationDetails.facilitySummaryListTable(0).bankReviewDateValue().should('contain', 'Required');
+      submitButton().should('not.exist');
+    });
+
+    it('bank review date page does not show pre-populated values', () => {
+      applicationDetails.facilitySummaryListTable(0).bankReviewDateAction().click();
+
+      bankReviewDate.bankReviewDateDay().should('have.value', '');
+      bankReviewDate.bankReviewDateMonth().should('have.value', '');
+      bankReviewDate.bankReviewDateYear().should('have.value', '');
+    });
+  });
+});
```

Comment:
Are we able to add an E2E test to check that eligibility has been updated to the latest version?

It would be ideal to check that all properties modified during the cloning are as expected in E2E

Reply:
I'm not exactly sure how I'd check that? Moreover it would fall under the existing `e2e-tests\gef\cypress\e2e\clone\clone-gef-deal.spec.js` test.

Reply:
Unless i'm missing something, it looks like we don't have full E2E coverage for GEF eligibility criiteria I.E criteria 5 should render X text 😞 could improve in the future.

Can we add some coverage for the other cloned fields though please i.e deal name etc?

Reply:
Bump, can we make sure we have tests for as many fields as possible (that are modified during cloning)?

https://github.com/UK-Export-Finance/dtfs2/pull/3782/files/1403404a951cb4617c6164e29e2c3cea47f2d415#diff-a467928018f518e2c357f605eab63127ac4b8e9d7e04dc1c8a2989a45d02cd13R66

Reply:
There's no real existing tests at all for the actual ECs that are used in a deal thats just been created, nevermind cloned.

(e.g e2e-tests\gef\cypress\e2e\eligibility\automatic-cover.spec.js just checks validation, e2e-tests\gef\cypress\e2e\eligibility\eligible-automatic-cover.spec.js doesnt check much at all)

Testing this in `clone-gef-deal.spec.js` would hugely increase the size of the PR needing me to work out what values should be visible & how to view them & then adding data-cy and new selectors to find these values on the screen. I don't think thats really within the scope of this ticket?

Reply:
OK, not ideal - can we create a ticket in the backlog please to add this E2E coverage?

We should also have at least 1x test to ensure that a deal can be submitted after cloning.

Reply:
Done https://ukef-dtfs.atlassian.net/browse/DTFS2-7610